python/cudf/cudf/tests/input_output/test_csv.py

```
@@ -2262,3 +2263,63 @@ def mock_size(*args):
+    cudf_output = df.to_csv(index=False, quoting=csv.QUOTE_MINIMAL)
+    pandas_output = pdf.to_csv(index=False, quoting=csv.QUOTE_MINIMAL)
+
+    assert cudf_output == pandas_output
+
+
+@pytest.mark.parametrize("quoting", [csv.QUOTE_ALL, csv.QUOTE_NONNUMERIC])
+def test_to_csv_quoting_unsupported(quoting):
+    """Test that unsupported quoting styles raise NotImplementedError."""
+    df = cudf.DataFrame({"a": [1, 2, 3], "b": ["x", "y", "z"]})
+    with pytest.raises(
+        NotImplementedError, match="quoting=.* is not supported"
```

Ruff (0.15.18)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)